File: Assets/Bundles/Shared/Resources/TS/UserInput/Drivers/PreferredDriver.lua
-- Compiled with unity-ts v2.1.0-75
local Signal = require("Shared/TS/Util/Signal").Signal
local PreferredDriver
do
	PreferredDriver = setmetatable({}, {
		__tostring = function()
			return "PreferredDriver"
		end,
	})
	PreferredDriver.__index = PreferredDriver
	function PreferredDriver.new(...)
		local self = setmetatable({}, PreferredDriver)
		return self:constructor(...) or self
	end
	function PreferredDriver:constructor()
		self.scheme = UserInputService.InputProxy:GetScheme()
		self.SchemeChanged = Signal.new()
		UserInputService.InputProxy:OnSchemeChangedEvent(function(scheme)
			self.scheme = scheme
			self.SchemeChanged:Fire(scheme)
		end)
	end
	function PreferredDriver:GetScheme()
		return self.scheme
	end
	function PreferredDriver:instance()
		if self.inst == nil then
			self.inst = PreferredDriver.new()
		end
		return self.inst
	end
end
return {
	PreferredDriver = PreferredDriver,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Bundles/@Easy/Core/Core~/src/Shared/UserInput/Drivers/PreferredDriver.ts
import { Signal } from "Shared/Util/Signal";

export class PreferredDriver {
	private static inst: PreferredDriver;

	private scheme = InputBridge.Instance.GetScheme();

	public readonly schemeChanged = new Signal<[scheme: string]>();

	private constructor() {
		// UserInputService.InputProxy.OnSchemeChangedEvent((scheme) => {
		// 	this.scheme = scheme;
		// 	this.SchemeChanged.Fire(scheme);
		// });
	}

	public GetScheme() {
		return this.scheme;
	}

	/** **NOTE:** Internal only. Use `Preferred` class instead. */
	public static Instance() {
		return (this.inst ??= new PreferredDriver());
	}
}
